lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -852,27 +844,35 @@ export class UserAgentApplication {
         this.cacheStorage.setItem(Constants.renewStatus + expectedState, Constants.tokenRenewStatusCancelled);
       }
     }, this.config.system.loadFrameTimeout);
+
+    const iframe = await this.loadFrame(urlNavigate, frameName);
+    const hash = await this.pollWindowForHash(iframe.contentWindow);
+    this.handleAuthenticationResponse(hash);
   }
 
   /**
    * @hidden
    * Loads iframe with authorization endpoint URL
    * @ignore
    */
-  private loadFrame(urlNavigate: string, frameName: string): void {
+  private loadFrame(urlNavigate: string, frameName: string): Promise<HTMLIFrameElement> {
```

Comment:
nit: [consistency] You can also use async await instead of `new Promise`.

Reply:
Need the new `Promise` since we need the `setTimeout`.